lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -1621,16 +1621,18 @@ namespace Js
 
     BOOL JavascriptProxy::GetDiagTypeString(StringBuilder<ArenaAllocator>* stringBuilder, ScriptContext* requestContext)
     {
+        const JavascriptProxy* proxy = UnwrapNestedProxies(this);
+
         //RecyclableObject* targetObj;
-        if (this->handler == nullptr)
+        if (proxy->handler == nullptr)
```

Comment:
wondering why we check for handler instead of target?

Reply:
We do this in other places. I assume that revoking a proxy removes the handler, so the presence of the handler is the indication of unrevoked proxy. 
Basically - once we have a handler, we must have a target. It may not be so otherwise, so checking for the handler is more reliable check for a "good" proxy.

Reply:
I do not know if such scenario exist, but we use ` if (handler== nullptr)` in a lot of places to check whether it is revoked and produce TypeError. That seems to imply that we should not unwrap proxies with `handler == null`, otherwise we would be losing those errors.

Are you suggesting that we should check target instead?
Is there a concern that we would be unwrapping revoked proxies instead of erroring on them?

Reply:
no need to change anything. I pointed that out as this in-fact looks inconsistent if someone just reads the code. I think a good cleanup (not necessarily you need to do with this change) is to change handler == null to IsRevoked(). That way in the IsRevoked function we could assert that target is null if the handler is null. 

---
In reply to: [205561914](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205561914) [](ancestors = 205561914)

Reply:
Yes, the part that the code just checks `handler == null` is indeed confusing. I am just trying to be consistent with that. Adding `IsRevoked()` would be a good change. 
I will enter an issue to follow up on that - to separate the refactoring/hygiene work from fixing a regression.

Reply:
Logged [https://github.com/Microsoft/ChakraCore/issues/5542](https://github.com/Microsoft/ChakraCore/issues/5542)   "Add an `IsRevoked` helper to JavascriptProxy for more code clarity"

---
In reply to: [205585921](https://github.com/Microsoft/ChakraCore/pull/5529#discussion_r205585921) [](ancestors = 205585921)